### PR TITLE
Add oneDNN/XNNPACK backend and support MobileNet/SqueezeNet model

### PR DESCRIPTION
**Pull request by fujunwei**
The commit supports 5 first-wave ops required by MobileNet/SequeezeNet example including Clamp, batchNormalization, LeakyRelu, Concat and Gemm.

It also adds XNNPACK and oneDNN backends that is mainly for the investigation of WebNN [Operation Level Execution
](https://webmachinelearning.github.io/webnn/#usecase-op-level-exec) use case. So only a limited set of tests (such as of conv2d) is expected to pass.

It also adds experimental sync build and compute API to implement  TF.js Wasm backend by WebNN API.

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -488,3 +488,257 @@ SOFTWARE AND DOCUMENTATION, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
 DAMAGE.
 
 ---------------------------------------------------------
+
```
Comment (huningxin) [inline NOTICE:491]: The XNNPACK license needs to be added into this doc.

File: DEPS [hunk 2]
```diff
@@ -11,13 +11,33 @@ vars = {
...
+  },
+
+  # Dependencies required for tests and examples.
+  'third_party/stb': {
+    'url': '{github_git}/nothings/stb@b42009b3b9d4ca35bc703f5310eedc74f584be58'
+  },
+  'node/third_party/webnn-polyfill': {
+    'url': '{github_git}/webmachinelearning/webnn-polyfill.git@5c5e9f6f5dbbe9dc67c448c8f7a55b6ddf4dee25'
+  },
+  'node/third_party/webnn-polyfill/test-data': {
```
Comment (huningxin) [inline DEPS:38]: As test-data is also used by cpp test cases, will it make sense to move it (probably together with webnn-polyfill) to top third_party?

File: NOTICE (hunk 1 above)
Comment (huningxin) [reply NOTICE:491]: And cnpy as well.

File: webnn.json [hunk 3]
```diff
@@ -415,7 +531,15 @@
           {"name": "userdata", "type": "void", "annotation": "*"},
           {"name": "outputs", "type": "named outputs", "optional": true}
         ]
+      },
+      {
+        "name": "computeSync",
```
Comment (huningxin) [inline webnn.json:536]: It looks like it introduces the experimental sync build and compute API. I guess it is for wasm lib investigation according to https://github.com/webmachinelearning/webnn/issues/156#issuecomment-846828170. It probably needs to be mentioned in the commit log and PR message.

Please note the PR for sync build and compute is under review: https://github.com/webmachinelearning/webnn/pull/174

**Review by huningxin — COMMENTED** [review]
Great work. Thanks much @fujunwei . The implementation would help the group for spec discussions, in particular for recent wasm op-level execution https://github.com/webmachinelearning/webnn/issues/156.

I left some comments for clarity. Please take a look.

**Comment by fujunwei** [comment]
Thanks, done. Please take a look again.

**Comment by BruceDai** [comment]
Thanks much @fujunwei!

I tested with this PR by both DML backend and OpenVINO backend, here's tests and examples PASS status table as following:

|Backend|End2EndTests (Pass / Total)|LeNet Example|SqueezeNet1.1 nchw Example|SqueezeNet1.0 nhwc Example|MobileNetV2 nchw Example|MobileNetV2 nhwc Example|
|----|----|----|----|----|----|----|
|DML|148 / 148|✔ | ✔ | ✔ | ✔ | ✔ |
|OpenVINO|146 / 148|✔ | ✔ | ✔ | ✔ | ✔ |

And two MaxPool2d tests with dilations options (**Pool2dTests.MaxPool2dDilationsDefault**, **Pool2dTests.MaxPool2dDilationsNhw**) failed when running OpenVINO backend. @fujunwei PTAL, thanks.

File: .gitignore [hunk 4]
```diff
@@ -91,3 +95,5 @@ $RECYCLE.BIN/
 ### Node ###
 node/generator/specifications
 node/PATH_TO_DAWN
+node/third_party/webnn-polyfill
```
Comment (huningxin) [inline .gitignore:98]: duplicated with L22?

**Review by huningxin — APPROVED** [review]
LGTM with a nit. Thanks @fujunwei for contributing this PR and @BruceDai for testing.

File: .gitignore (hunk 4 above)
Comment (fujunwei) [reply .gitignore:97]: Removed it, thanks.

**Comment by BruceDai** [comment]
Thanks much @fujunwei for kindly explanation!

**Comment by huningxin** [comment]
> It also adds XNNPACK and oneDNN backends that is mainly for the investigation of WebNN [Operation Level Execution ](https://webmachinelearning.github.io/webnn/#usecase-op-level-exec) use case. So only a limited set of tests (such as of conv2d) is expected to pass.
> 
> It also adds experimental sync build and compute API to implement TF.js Wasm backend by WebNN API.

Given [the PR of sync API](https://github.com/webmachinelearning/webnn/pull/174) was merged and  [the investigation of wasm op-level execution](https://github.com/webmachinelearning/webnn/issues/156#issuecomment-846828170) was done, I am proposing to merge this PR to allow participants to reproduce the results. And it would also unblock the development for #6 and #12 .

Any objections? @anssiko @wchao1115 @pyu10055

**Comment by anssiko** [comment]
SGTM!

**Pull request merged**